frontend/src/modules/equipment/components/EquipmentTable.tsx

```diff
@@ -0,0 +1,218 @@
+
+interface EquipmentTableProps {
+  refresh: boolean;
+}
+
+export const EquipmentTable = ({ refresh }: EquipmentTableProps) => {
+  const api = useContext(ApiContext);
+  const toast = useToast();
+
+  const [equipment, setEquipment] = useState<EquipmentDTO[] | undefined>([]);
+  const equipmentToEdit = useRef<EquipmentDTO | null>(null);
+  const equipmentToDelete = useRef<EquipmentDTO | null>(null);
+
+  const {
+    onOpen: onEditModalOpen,
+    isOpen: isEditModalOpen,
+    onClose: onEditModalClose,
+  } = useDisclosure();
+
+  const {
+    isOpen: isDeleteDialogOpen,
+    onOpen: onDeleteDialogOpen,
+    onClose: onDeleteDialogClose,
+  } = useDisclosure();
+
+  const fetchEquipment = useMutation(() => api.getAllEquipments(), {
+    onSuccess: (data) => {
+      setEquipment(() => data.data);
+    },
+  });
+
+  const deleteEquipment = useMutation(
+    (equipmentDTO: EquipmentDTO) =>
+      api.deleteEquipment(equipmentDTO.id as string),
+    {
+      onSuccess: () => {
+        fetchAllEquipment();
+      },
+    }
+  );
+
+  const fetchAllEquipment = () => {
+    fetchEquipment.mutate();
+  };
+
+  useEffect(() => {
+    fetchAllEquipment();
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [refresh]);
+
+  const handleEdit = (equipmentDTO: EquipmentDTO) => {
+    onEditModalOpen();
+    equipmentToEdit.current = equipmentDTO;
+  };
+
+  const handleDelete = (equipmentDTO: EquipmentDTO) => {
+    onDeleteDialogOpen();
+    equipmentToDelete.current = equipmentDTO;
+  };
+
+  const handleDeleteConfirm = async () => {
+    const equipmentDTO = equipmentToDelete.current;
+
+    if (equipmentDTO === null) {
+      return;
+    }
+
+    await deleteEquipment.mutateAsync(equipmentDTO);
+    toast({ title: 'Deleted equipment successfully' });
+    onDeleteDialogClose();
+  };
+
+  const handleEditComplete = () => {
+    fetchAllEquipment();
+    onEditModalClose();
+  };
+
+  if (fetchEquipment.isLoading) {
+    return (
+      <Center minH={80} minW={80}>
+        <Spinner colorScheme='brand' />
+      </Center>
+    );
+  }
+
+  return (
+    <TableContainer position='relative' bg='foreground' p={0} m={0}>
+      {equipmentToEdit.current && (
+        <Modal
+          title='Equipment Edit'
+          isOpen={isEditModalOpen}
+          onClose={onEditModalClose}
+        >
+          <EquipmentEditForm
+            initialEquipment={equipmentToEdit.current}
+            onComplete={handleEditComplete}
+          />
+        </Modal>
+      )}
+
+      <Dialog
+        title='Delete'
+        description='Proceeding with this action cannot be undone. Continue?'
+        data={equipmentToDelete.current?.name.toString()}
+        isOpen={isDeleteDialogOpen}
+        onClose={onDeleteDialogClose}
+        onConfirm={handleDeleteConfirm}
+      />
+      <Table colorScheme='gray' fontSize='sm' variant='simple' p={0} m={0}>
+        <Thead>
+          <Tr>
+            <Th>
+              {/* <Icon as={FaUserAlt} mr={2} /> */}
```

Comment:
We will be having icons on all the column headers, soon when we have the design. But for now, I think it would be nice to just comment on them instead of having unofficial icons.